### ISSUE TITLE
finds __class__ undefined

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#569]
In GitLab by @graingert on Mar 22, 2017, 12:31

*Please describe how you installed Flake8*

Example:

```
python3 -m venv ~/.virtualenvs/flake8
workon flake8
pip install -U pip setuptools wheel
pip install flake8 flake8-commas
```

*Please provide the exact, unmodified output of `flake8 --bug-report`*

nah

*Please describe the problem or feature*
flake8 thinks that `__class__` is undefined in:

```python
class Option:
    def __init__(self, value):
        self._value = value

    def map(self, fn):
        if self._value is not None:
            return __class__(fn(self._value))  # <-- undefined, when it's clearly defined
        return self

    def get(self):
        if self._value is not None:
            return self._value
        raise ValueError('Noooo')
```

*If this is a bug report, please explain with examples (and example code) what you expected to happen and what actually happened.*

I expected __class__ to be considered defined, because it is.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Mar 22, 2017, 12:54

closed

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Mar 22, 2017, 12:55

This is being reported by PyFlakes and thus needs to be reported there.

See also: https://bugs.launchpad.net/pyflakes/+bug/1487725
